iotile_analytics_interactive/iotile_analytics/interactive/reports/cloud_uploader.py

```diff
@@ -243,7 +243,7 @@ def _escape_quote(indata):
     body = b'\r\n'.join(lines)
 
     headers = {
-        b'Content-Type': b'multipart/form-data; boundary={0}'.format(boundary.encode('utf-8'))
+        b'Content-Type': 'multipart/form-data; boundary={0}'.format(boundary).encode('utf-8')
```

Comment:
Shouldn't the key (Content-Type) also be a regular string?

Reply:
In this case no, unfortunately.  It's a bit of a tightrope.  Python 2 wants str but since this file has `unicode_literals` imported to make sure all the messages are properly unicode the `b''` forces normal strings on python 2 since `str` and `bytes` are the same there.  Then `_generic_url_poster` upgrades the bytes to a `unicode` string on python 3.  

Technically the original (before the change) is what we want but apparently python 3 doesn't support `.format` on bytes so I needed to do the conversion from unicode to bytes after the `format`.